jetty-ee10/jetty-ee10-servlet/src/test/java/org/eclipse/jetty/ee10/servlet/DefaultServletCombinationsTest.java

```diff
@@ -117,44 +118,79 @@ public void init() throws Exception
 
     private void startServer(boolean pathInfoOnly, ResourceService.WelcomeMode welcomeMode) throws Exception
     {
-        ServletHolder defaultHolder = context.addServlet(DefaultServlet.class, "/static/*");
-        defaultHolder.setInitParameter("pathInfoOnly", String.valueOf(pathInfoOnly));
+        ServletHolder defaultHolder;
+        if (pathInfoOnly)
+        {
+            defaultHolder = context.addServlet(DefaultServlet.class, "/static/*");
+            defaultHolder.setInitParameter("pathInfoOnly", "true");
+            context.addServlet(TeapotServlet.class,  "/");
+        }
+        else
+        {
+            defaultHolder = context.addServlet(DefaultServlet.class, "/");
+            defaultHolder.setInitParameter("pathInfoOnly", "false");
+        }
```

Comment:
This looks good, but we are still missing the scenario of a `DefaultServelt` at non root path, but with `pathInfoOnly` false.  Not sure this is a good configuration to support?

I wonder if we just get rid of the configuration and always act as `pathInfoOnly`, so if req.getHttpServletMapping().getMappingMatch() == MappingMatch.DEFAULT, then we use the servletPath, otherwise we use the pathInfo.   Is that simpler? Is that sufficient?

Reply:
I've added the _DefaultServlet at non root path_ case as an extra test method as I found it confusing to mix these combinations with the _DefaultServlet at root path_ ones.

Eliminating one parameter is tempting, but is your suggestion spec-compliant?

Reply:
The spec says nothing about non-default servlet.   It just says the default default servlet should server resources, but may be replaced by an application provided default servlet that may do anything.
It is entirely a jetty feature that we let our `DefaultServlet` be used in non-default ways (It should have been called `ResourceServlet`).

So spec is of no concern here.

Only concern is if there is a valid use-case to have our DefaultServlet` deployed as non-default, but using the full servletPath + pathInfo

Reply:
You're probably onto something, but that seems to be too large for the scope of this PR. Or this PR should be moved to after the 12.0.0 release.